### ISSUE TITLE
question about taint tracking

### ISSUE THREAD

**Issue by wuyifan18** [xlab-uiuc/cdep-fse#1]
hi @tianyin, thanks for the great work. 
I want to know whether the taint tracking is based on open source or self-developed?

**Comment by tianyin** [comment]
I guess you are asking whether we used some open-source tools or implemented ourselves? 

It's the latter. @chenqingrong wrote some very simple (likely overly simplified) static taint tracking:
https://github.com/xlab-uiuc/cdep-fse/blob/master/src/main/java/dataflow/IntraProcedureAnalysis.java

There are other tools if you are looking for other static taint tracking frameworks, such as FlowDroid and Semmle (I never used Semmle before but @aastorg2 and @thakkarparth007 explored it).

We did use TaintDroid but was affected by the limitation of its non-deterministic path reconstruction (@X-czh can tell more). So, @X-czh and @zjzijielu quickly hacked a static taint tracker for config values (https://github.com/xlab-uiuc/cflow); not sure if it helps. My understanding is that if you only care about the taints rather than the paths, FlowDroid is a reasonable tool.
